src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -888,17 +907,21 @@ private static Symbol ComputeImplementationForInterfaceMember(Symbol interfaceMe
             if ((object)implicitImpl == null && seenTypeDeclaringInterface)
             {
                 // Check for default interface implementations
-                if (!interfaceMember.IsAbstract)
-                {
-                    implicitImpl = interfaceMember;
-                }
+                implicitImpl = FindMostSpecificImplementation(interfaceMember, implementingType, ref useSiteDiagnostics, diagnostics);
+            }
+
+#if !DEBUG
```

Comment:
I didn't understand why the use-site diagnostics should only be produced with debug compiler bits. Maybe add a comment?